src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -70,6 +84,76 @@ public static void initIndexes() throws Exception {
         Collections.sort(indexes);
     }
     
+    @AfterClass
+    public static void tearDownStatics() {
+        indexes = null;
+        indicesDir = null;
+    }
+    
+    @Override
+    public Settings nodeSettings(int ord) {
+        return ImmutableSettings.builder()
+            .put(Node.HTTP_ENABLED, true) // for _upgrade
+            .put(MergePolicyModule.MERGE_POLICY_TYPE_KEY, NoMergePolicyProvider.class) // disable merging so no segments will be upgraded
+            .build();
+    }
+    
+    void setupDataLoadingNodes() throws Exception {
+        Path dataDir = newTempDirPath(LifecycleScope.SUITE);
+        indicesDir = dataDir.resolve(internalCluster().getClusterName()).resolve("nodes/0/indices/");
+        assertFalse(Files.exists(indicesDir));
+        Files.createDirectories(indicesDir);
+        ImmutableSettings.Builder nodeSettings = ImmutableSettings.builder()
+            .put("path.data", dataDir.toAbsolutePath());
+        internalCluster().startNode(nodeSettings);
+    }
+
+    String loadIndex(String indexFile) throws Exception {
+        Path unzipDir = newTempDirPath();
+        Path unzipDataDir = unzipDir.resolve("data");
+        String indexDir = indexFile.replace(".zip", "");
+        String indexName = indexDir.toLowerCase(Locale.ROOT);
```

Comment:
Index names cannot contain upper case letters, but the zip file names are based on the elasticsearch version, which can contain `Beta` and `RC`.